torch/csrc/distributed/rpc/init.cpp

```diff
@@ -149,7 +149,10 @@ PyObject* rpc_init(PyObject* /* unused */) {
               >>> # count is automatically updated.
               >>> rpc.rpc_sync("worker1", f, args(rref,))
           )")
-          .def(py::init<const py::object&>())
+          .def(
+              py::init<const py::object&, const py::object&>(),
+              py::arg("value"),
+              py::arg("type_hint") = py::none())
```

Comment:
Do we want to expose API doc for this or this will this stay as an internal feature for now?

Reply:
We might no want to expose this `type_hint` field in Python.
It is possible that we handle it in c++?
The only known use case is to get around TorchScript limitation that most users should not care.

Reply:
Let's keep it private for now. 

This hidden default arg might be removed, since one alternative is to infer the type automatically for users. So, at this moment, I can't precisely foreseen what following RRef JIT typing changes we will make to it.

Reply:
@zzzwen This will bind to Python RRef.__init__.